### PR TITLE
Feat/feed helpers

### LINKED ISSUE #535: Edit Feed

### PR DESCRIPTION
**Pull request by radical-ube**
resolves #535 

this should be last thing needed after the crud PR to bring edit feeds to parity. this PR adds getter/setter methods to address the `audio_format` and `include_zones` fields.

### CODE REVIEW

File: app/javascript/controllers/toggle_field_controller.js
Comment (radical-ube) [inline app/javascript/controllers/toggle_field_controller.js:25]: this small change is to account for being able to handle display toggling for multiple types. in this case, `mp3`, `wav`, or `flac`.

File: app/views/feeds/_form_main.html.erb
Comment (radical-ube) [inline app/views/feeds/_form_main.html.erb:94]: for this and the below fields, 2 things are defined on this div line: initial display class, and display toggle.

the initial display is in `feeds_helper` and is based on `audio_format[:f]` or `audio_format.blank?` depending on the field.

the toggle is triggered by the change in value of `audio_type` (aka `audio_format[:f]`). so this one displays only with `mp3`, bitdepth with either `wav` or `flac`, and the other two with any of the three types. and `d-none` for blank.

File: app/views/feeds/_form_custom_fields.html.erb
Comment (cavis) [inline app/views/feeds/_form_custom_fields.html.erb:86]: Where did this `System (Sonic IDs)` label come from?  Is that in the designs?  I've never heard a sonic_id zone referred to as "system" before... can't decide how confusing I find it yet.  (That term is overridden by Campaign Type in Augury).

File: app/controllers/feeds_controller.rb
Comment (cavis) [inline app/controllers/feeds_controller.rb:97]: Unrelated to this PR request: can you sort the Feeds tabs by something?  Maybe just `created_at: :asc`, so they show in the order you created them?  Just a bit jarring to compare with Publish when they're in a different order.

File: app/models/concerns/feed_ad_zone.rb
Comment (cavis) [reply app/models/concerns/feed_ad_zone.rb:21]: And for my 3rd comment in this thread...

Maybe it's better if we deviate this UI from Publish?  Publish doesn't make a ton of sense anyways.  So for the allowed values of `include_zones`:

- When `nil` ... show all 4 boxes checked.  And when the user saves with all 4 checked, set to nil.
- When `[]` ... show all 4 boxes unchecked.
- When `["<name>"]` ... check just those 1, 2, or 3 zones.

That make more sense?  It maintains the existing data as well (it can't be an array with 4 items in it ... it becomes nil instead).

**Review by cavis — CHANGES_REQUESTED** [review]
Okay, generally works good, a few suggestions/comments, and 1 gotcha that needs to be fixed.

I think after these 2 PRs merge, we should do a bit of QA and figure out where this is on parity with Publish.

File: app/views/feeds/_form_custom_fields.html.erb
Comment (radical-ube) [reply app/views/feeds/_form_custom_fields.html.erb:86]: yes, i think i pulled verbiage from the design.

File: app/models/concerns/feed_audio_format.rb
Comment (radical-ube) [reply app/models/concerns/feed_audio_format.rb:55]: tests added below!

File: app/views/feeds/_form_custom_fields.html.erb
Comment (radical-ube) [reply app/views/feeds/_form_custom_fields.html.erb:86]: i can change the labels back to parity with Publish and we can bring it up it later.

File: test/models/concerns/feed_ad_zone_test.rb
Comment (radical-ube) [inline test/models/concerns/feed_ad_zone_test.rb:56]: this suite and `#remove_zone` are possibly misnomers? since i don't test either of those methods directly (they're both private), but rather through calling each of the `method=` setters. so i'm not sure if i should have gone the more redundant route and just had each of those be their own suites and repeating the same tests in each? thinking about long-term readability if this is the kind of thing that would be confusing to understand a year from now.

Comment (cavis) [reply test/models/concerns/feed_ad_zone_test.rb:56]: Ah, I see what you're saying.

I think either way is fine - though it is a bit odd to reference a private method here, even as a comment.  Stylistically, I might prefer the more verbose style?  Or write one mega-test, and just loop over all 4 zone names, running the same checks?

Your call.  Definitely multiple ways.

File: test/models/concerns/feed_ad_zone_test.rb
Comment (cavis) [inline test/models/concerns/feed_ad_zone_test.rb:65]: Why the parenths around all these?

**Review by cavis — APPROVED** [review]
👍 looks good to me.

File: test/models/concerns/feed_ad_zone_test.rb
Comment (radical-ube) [reply test/models/concerns/feed_ad_zone_test.rb]: ended up rewriting the verbose way and i think it reads better.

**Pull request merged**
